lib/v8-to-istanbul.js

```diff
@@ -0,0 +1,178 @@
+const assert = require('assert')
+const convertSourceMap = require('convert-source-map')
+const CovBranch = require('./branch')
+const CovFunction = require('./function')
+const CovSource = require('./source')
+const { dirname, resolve } = require('path')
+const { readFileSync } = require('fs')
+const { SourceMapConsumer } = require('source-map')
+
+const isNode10 = !!process.version.match(/^v10/)
+
+// Injected when Node.js is loading script into isolate pre Node 11.
+// see: https://github.com/nodejs/node/pull/21573.
+const cjsWrapperLength = isNode10 ? require('module').wrapper[0].length : 0
+
+module.exports = class V8ToIstanbul {
+  constructor (scriptPath, wrapperLength) {
+    assert(typeof scriptPath === 'string', 'scriptPath must be a string')
+    this.path = parsePath(scriptPath)
+    this.wrapperLength = wrapperLength === undefined ? cjsWrapperLength : wrapperLength
+    this.generatedLines = []
+    this.branches = []
+    this.functions = []
+    this.sourceMap = undefined
+    this.source = undefined
+    this.sourceTranspiled = undefined
+  }
+  async load () {
+    const rawSource = readFileSync(this.path, 'utf8')
+    // if we find a source-map (either inline, or a .map file) we load
+    // both the transpiled and original source, both of which are used during
+    // the backflips we perform to remap absolute to relative positions.
+    const rawSourceMap = convertSourceMap.fromSource(rawSource) || convertSourceMap.fromMapFileSource(rawSource, dirname(this.path))
+    if (rawSourceMap) {
+      this.path = resolve(dirname(this.path), rawSourceMap.sourcemap.sources[0])
```

Comment:
This only accounts for the first source file when there can be many. For example I'm working with a file that has 259 sources in it.

Reply:
@Friss it also doesn't account for relative vs., absolute paths.

I think it might be worth pulling in this module from Istanbuljs:

Reply:
Yeah we build our apps with webpack and has the babel toolchain etc.